api/src/test/java/org/openmrs/module/ModuleTest.java

```diff
@@ -0,0 +1,127 @@
+
+		mockModule = ModuleFactory.getModuleById("test1");
+	}
+
+	/*
+	 * @verifies not expand extensionNames if extensionNames is null
+	 * @see Module#getExtensions()
+	 */
+	@Test
+	public void getExtensions_shouldNotExpandExtensionNamesIfExtensionNamesIsNull() {
+		ArrayList<Extension> extensions = new ArrayList<Extension>();
+
+		Extension mockExtension = new MockExtension();
+		extensions.add(mockExtension);
+
+		mockModule.setExtensions(extensions);
+		mockModule.setExtensionNames(null);
+		ArrayList<Extension> ret = new ArrayList<Extension>(mockModule.getExtensions());
+
+		assertEquals(extensions.get(0), ret.get(0));
+		assertEquals(1, ret.size());
+	}
+
+	/*
+         * @verifies not expand extensionNames if extensionNames is empty
+	 * @see Module#getExtensions()
+	 */
+	@Test
+	public void getExtensions_shouldNotExpandExtensionNamesIfExtensionNamesIsEmpty() {
+		ArrayList<Extension> extensions = new ArrayList<Extension>();
+
+		Extension mockExtension = new MockExtension();
+		extensions.add(mockExtension);
+
+		mockModule.setExtensions(extensions);
+		mockModule.setExtensionNames(new IdentityHashMap<String, String>());
+		ArrayList<Extension> ret = new ArrayList<Extension>(mockModule.getExtensions());
+
+		assertEquals(extensions.get(0), ret.get(0));
+		assertEquals(1, ret.size());
+	}
+
+	/*
+         * @verifies not expand extensionNames if extensions matches extensionNames
+	 * @see Module#getExtensions()
+	 */
+	@Test
+	public void getExtensions_shouldNotExpandExtensionNamesIfExtensionsMatchesExtensionNames() {
+		ArrayList<Extension> extensions = new ArrayList<Extension>();
+		IdentityHashMap<String, String> extensionNames = new IdentityHashMap<String, String>();
+
+		Extension mockExtension = new MockExtension();
+		mockExtension.setPointId("1");
+		extensions.add(mockExtension);
+		extensionNames.put("1", mockExtension.getClass().getName());
+
+		mockModule.setExtensions(extensions);
+		mockModule.setExtensionNames(extensionNames);
+		ArrayList<Extension> ret = new ArrayList<Extension>(mockModule.getExtensions());
+
+		assertEquals(extensions.get(0), ret.get(0));
+		assertEquals(1, ret.size());
+	}
+
+	/*
+         * @verifies expand extensionNames if extensions does not match extensionNames
+	 * @see Module#getExtensions()
+	 */
+	@Test
+	public void getExtensions_shouldExpandExtensionNamesIfExtensionsDoesNotMatchExtensionNames() {
+		ArrayList<Extension> extensions = new ArrayList<Extension>();
+		IdentityHashMap<String, String> extensionNames = new IdentityHashMap<String, String>();
+
+		Extension mockExtension = new MockExtension();
+		mockExtension.setPointId("1");
+		extensions.add(mockExtension);
+		extensionNames.put("2", mockExtension.getClass().getName());
+
+		mockModule.setExtensions(extensions);
+		ArrayList<Extension> ret = new ArrayList<Extension>(mockModule.getExtensions());
+
+		//getExtensionNames will return an empty extensions list since MockExtension won't
+		//have a valid class loader
```

Comment:
According to this comment, this assertion has nothing to do with what this test. In other words, the assertion would pass if MockExtension had a valid class loader.